src/icarus/userinterface.py

```diff
@@ -0,0 +1,170 @@
+from tkinter import Tk, Label, Entry, Button, Frame, Checkbutton, BooleanVar, StringVar
+from pathlib import Path
+
+class UserInterface:
+    """Provides a user interface using tkinter to allow the user to control aspects of the modelling process,
+        including selecting parameters to be perturbed and defining their values and how many validation datasets
+        to generate for each parameter.
+    """
+    def __init__(self):
+        pass
+
+
+    def accept_parameters(self, parameters: dict[str, float]) -> tuple[list[int], dict[str, list[float]]]:
+        """accept_parameters: used to allow user to select which parameters to perturb, and the range,
+            interval of perturbation values, and number of validation values to be used for each parameter.
```

Comment:
The indentation on many of the docstrings is causing them to render incorrectly. Be sure to drop back in line with the opening `"""` on each line.
```suggestion
        """accept_parameters: used to allow user to select which parameters to perturb, and the range,
        interval of perturbation values, and number of validation values to be used for each parameter.
```